File: ios/Sasquatsh/Views/Events/GameSystemForms/Warhammer40kConfigFormSections.swift
import SwiftUI

struct Warhammer40kConfigFormSections: View {
    @Binding var config: Warhammer40kConfigState

    var body: some View {
        gameSetupSection
        missionSection
        armyRulesSection
        terrainSection
        eventStructureSection
        if config.showCrusadeSettings {
            crusadeSection
        }
        if config.showArmyListSubmission {
            armyListSection
        }
        prizesSection
    }

    // MARK: - Game Setup

    private var gameSetupSection: some View {
        Section("Game Setup") {
            Picker("Game Type", selection: Binding(
                get: { config.gameType ?? "matched" },
                set: { config.gameType = $0 }
            )) {
                Text("Matched Play").tag("matched")
                Text("Narrative").tag("narrative")
                Text("Crusade").tag("crusade")
                Text("Open Play").tag("open")
            }

            VStack(alignment: .leading, spacing: 8) {
                Text("Points Limit")
                    .font(.md3LabelMedium)
                    .foregroundStyle(Color.md3OnSurfaceVariant)

                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(spacing: 8) {
                        ForEach([500, 1000, 1500, 2000, 2500, 3000], id: \.self) { pts in
                            Button {
                                config.pointsLimit = pts
                                config.applyPointsDefaults()
                            } label: {
                                Text("\(pts)")
                                    .font(.md3LabelMedium)
                                    .padding(.horizontal, 12)
                                    .padding(.vertical, 6)
                                    .background(config.pointsLimit == pts ? Color.md3Primary : Color.md3SurfaceContainerHigh)
                                    .foregroundStyle(config.pointsLimit == pts ? Color.md3OnPrimary : Color.md3OnSurface)
                                    .clipShape(RoundedRectangle(cornerRadius: MD3Shape.small))
                            }
                        }
                    }
                }

                Stepper("Points: \(config.pointsLimit)", value: $config.pointsLimit, in: 100...5000, step: 100)
                    .onChange(of: config.pointsLimit) { _, _ in
                        config.applyPointsDefaults()
                    }
            }

            Picker("Player Mode", selection: $config.playerMode) {
                Text("1v1").tag("1v1")
                Text("2v2").tag("2v2")
                Text("Group").tag("group")
            }
            .pickerStyle(.segmented)
        }
    }

    // MARK: - Mission

    private var missionSection: some View {
        Section("Mission") {
            Picker("Mission Pack", selection: $config.missionPack) {
                Text("None").tag("")
                Text("Leviathan").tag("leviathan")
                Text("Pariah Nexus").tag("pariah_nexus")
                Text("Chapter Approved").tag("chapter_approved")
                Text("Crusade").tag("crusade")
                Text("Custom").tag("custom")
            }

            if !config.missionPack.isEmpty {
                Picker("Mission Selection", selection: Binding(
                    get: { config.missionSelection ?? "random" },
                    set: { config.missionSelection = $0 }
                )) {
                    Text("Random").tag("random")
                    Text("Pre-Selected").tag("pre_selected")
                }
            }

            TextField("Mission Notes", text: $config.missionNotes, axis: .vertical)
                .lineLimit(2...4)
        }
    }

    // MARK: - Army Rules

    private var armyRulesSection: some View {
        Section("Army Rules") {
            Toggle("Battle Ready Required", isOn: $config.battleReadyRequired)
            Toggle("WYSIWYG Required", isOn: $config.wysiwygRequired)
            Toggle("Forge World Allowed", isOn: $config.forgeWorldAllowed)
            Toggle("Legends Allowed", isOn: $config.legendsAllowed)
            Toggle("Allow Proxies", isOn: $config.allowProxies)

            if config.allowProxies {
                TextField("Proxy Notes", text: $config.proxyNotes, axis: .vertical)
                    .lineLimit(2...3)
            }

            TextField("Army Rules Notes", text: $config.armyRulesNotes, axis: .vertical)
                .lineLimit(2...4)
        }
    }

    // MARK: - Terrain & Table

    private var terrainSection: some View {
        Section("Terrain & Table") {
            Picker("Terrain Type", selection: $config.terrainType) {
                Text("Tournament Standard").tag("tournament")
                Text("Casual").tag("casual")
                Text("Bring Your Own").tag("bring_your_own")
            }

            Picker("Table Size", selection: $config.tableSize) {
                Text("44\" x 30\" (Combat Patrol)").tag("44x30")
                Text("44\" x 60\" (Standard)").tag("44x60")
                Text("Custom").tag("custom")
            }
        }
    }

    // MARK: - Event Structure

    private var eventStructureSection: some View {
        Section("Event Structure") {
            Picker("Event Type", selection: Binding(
                get: { config.eventType ?? "casual" },
                set: { config.eventType = $0 }
            )) {
                Text("Casual").tag("casual")
                Text("Tournament").tag("tournament")
                Text("Campaign").tag("campaign")
                Text("League").tag("league")
            }

            if config.eventType == "tournament" || config.eventType == "league" {
                Picker("Tournament Style", selection: Binding(
                    get: { config.tournamentStyle ?? "swiss" },
                    set: { config.tournamentStyle = $0 }
                )) {
                    Text("Swiss").tag("swiss")
                    Text("Single Elimination").tag("single_elimination")
                    Text("Round Robin").tag("round_robin")
                }

                HStack {
                    Text("Rounds")
                    Spacer()
                    TextField("Auto", value: $config.roundsCount, format: .number)
                        .keyboardType(.numberPad)
                        .multilineTextAlignment(.trailing)
                        .frame(width: 80)
                }

                Stepper("Round Time: \(config.roundTimeMinutes) min", value: $config.roundTimeMinutes, in: 30...300, step: 15)

                Toggle("Include Top Cut", isOn: $config.includeTopCut)

                Picker("Scoring", selection: Binding(
                    get: { config.scoringType ?? "win_loss" },
                    set: { config.scoringType = $0 }
                )) {
                    Text("Win/Loss").tag("win_loss")
                    Text("Win/Draw/Loss").tag("win_draw_loss")
                    Text("Battle Points").tag("battle_points")
                }
            }
        }
    }

    // MARK: - Crusade

    private var crusadeSection: some View {
        Section("Crusade Settings") {
            Stepper("Starting Supply Limit: \(config.startingSupplyLimit)", value: $config.startingSupplyLimit, in: 500...3000, step: 250)
            Stepper("Starting Crusade Points: \(config.startingCrusadePoints)", value: $config.startingCrusadePoints, in: 0...20)

            TextField("Progression Notes", text: $config.crusadeProgressionNotes, axis: .vertical)
                .lineLimit(2...4)
        }
    }

    // MARK: - Army List Submission

    private var armyListSection: some View {
        Section("Army List Submission") {
            Toggle("Require Army List", isOn: $config.requireArmyList)

            if config.requireArmyList {
                DatePicker("Submission Deadline",
                    selection: Binding(
                        get: { config.armyListDeadline ?? Date() },
                        set: { config.armyListDeadline = $0 }
                    ),
                    displayedComponents: [.date, .hourAndMinute]
                )

                TextField("Submission Notes", text: $config.armyListNotes, axis: .vertical)
                    .lineLimit(2...3)
            }
        }
    }

    // MARK: - Prizes

    private var prizesSection: some View {
        Section("Prizes & Entry") {
            Toggle("Has Prizes", isOn: $config.hasPrizes)

            if config.hasPrizes {
                TextField("Prize Structure", text: $config.prizeStructure, axis: .vertical)
                    .lineLimit(2...4)
            }

            HStack {
                Text("Entry Fee")
                Spacer()
                TextField("Free", text: $config.entryFee)
                    .keyboardType(.decimalPad)
                    .multilineTextAlignment(.trailing)
                    .frame(width: 80)
                Picker("", selection: $config.entryFeeCurrency) {
                    Text("USD").tag("USD")
                    Text("EUR").tag("EUR")
                    Text("GBP").tag("GBP")
                    Text("CAD").tag("CAD")
                }
                .frame(width: 70)
            }

            Toggle("Allow Spectators", isOn: $config.allowSpectators)
        }
    }
}

File: android/app/src/main/java/com/sasquatsh/app/views/events/Warhammer40kConfigFormSections.kt
package com.sasquatsh.app.views.events

import androidx.compose.animation.AnimatedVisibility
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.ExperimentalLayoutApi
import androidx.compose.foundation.layout.FlowRow
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.material3.DropdownMenuItem
import androidx.compose.material3.ExperimentalMaterial3Api
import androidx.compose.material3.ExposedDropdownMenuBox
import androidx.compose.material3.ExposedDropdownMenuDefaults
import androidx.compose.material3.FilterChip
import androidx.compose.material3.FilterChipDefaults
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.MenuAnchorType
import androidx.compose.material3.OutlinedTextField
import androidx.compose.material3.SegmentedButton
import androidx.compose.material3.SegmentedButtonDefaults
import androidx.compose.material3.SingleChoiceSegmentedButtonRow
import androidx.compose.material3.Slider
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import com.sasquatsh.app.models.Warhammer40kConfigState

@OptIn(ExperimentalMaterial3Api::class, ExperimentalLayoutApi::class)
@Composable
fun Warhammer40kConfigFormSections(
    config: Warhammer40kConfigState,
    onConfigChange: (Warhammer40kConfigState) -> Unit
) {
    // ─── Game Setup Section ───
    SectionHeader("Game Setup")

    // Game Type
    var gameTypeExpanded by remember { mutableStateOf(false) }
    val gameTypes = listOf(
        "matched" to "Matched Play", "narrative" to "Narrative",
        "crusade" to "Crusade", "open" to "Open Play"
    )
    ExposedDropdownMenuBox(expanded = gameTypeExpanded, onExpandedChange = { gameTypeExpanded = it }) {
        OutlinedTextField(
            value = gameTypes.firstOrNull { it.first == (config.gameType ?: "matched") }?.second ?: "Matched Play",
            onValueChange = {},
            readOnly = true,
            label = { Text("Game Type") },
            trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(gameTypeExpanded) },
            modifier = Modifier.fillMaxWidth().menuAnchor(MenuAnchorType.PrimaryNotEditable)
        )
        ExposedDropdownMenu(expanded = gameTypeExpanded, onDismissRequest = { gameTypeExpanded = false }) {
            gameTypes.forEach { (value, label) ->
                DropdownMenuItem(
                    text = { Text(label) },
                    onClick = {
                        onConfigChange(config.copy(gameType = value))
                        gameTypeExpanded = false
                    }
                )
            }
        }
    }

    Spacer(Modifier.height(8.dp))

    // Points Limit - quick-select chips
    Text("Points Limit", style = MaterialTheme.typography.labelMedium,
        color = MaterialTheme.colorScheme.onSurfaceVariant)
    Spacer(Modifier.height(4.dp))

    FlowRow(
        horizontalArrangement = Arrangement.spacedBy(8.dp),
        verticalArrangement = Arrangement.spacedBy(4.dp)
    ) {
        listOf(500, 1000, 1500, 2000, 2500, 3000).forEach { pts ->
            FilterChip(
                selected = config.pointsLimit == pts,
                onClick = { onConfigChange(config.copy(pointsLimit = pts)) },
                label = { Text("$pts") },
                colors = FilterChipDefaults.filterChipColors(
                    selectedContainerColor = MaterialTheme.colorScheme.primary,
                    selectedLabelColor = MaterialTheme.colorScheme.onPrimary
                )
            )
        }
    }

    Spacer(Modifier.height(4.dp))
    val pointsVal = (config.pointsLimit ?: 2000).toFloat()
    Text("Points: ${pointsVal.toInt()}", style = MaterialTheme.typography.bodyMedium)
    Slider(
        value = pointsVal,
        onValueChange = { onConfigChange(config.copy(pointsLimit = (it / 100).toInt() * 100)) },
        valueRange = 100f..5000f
    )

    Spacer(Modifier.height(8.dp))

    // Player Mode segmented
    SingleChoiceSegmentedButtonRow(modifier = Modifier.fillMaxWidth()) {
        val playerModes = listOf("1v1" to "1v1", "2v2" to "2v2", "group" to "Group")
        playerModes.forEachIndexed { index, (value, label) ->
            SegmentedButton(
                selected = (config.playerMode ?: "1v1") == value,
                onClick = { onConfigChange(config.copy(playerMode = value)) },
                shape = SegmentedButtonDefaults.itemShape(index, playerModes.size)
            ) {
                Text(label)
            }
        }
    }

    // ─── Mission Section ───
    Spacer(Modifier.height(16.dp))
    SectionHeader("Mission")

    var missionPackExpanded by remember { mutableStateOf(false) }
    val missionPacks = listOf(
        "" to "None", "leviathan" to "Leviathan", "pariah_nexus" to "Pariah Nexus",
        "chapter_approved" to "Chapter Approved", "crusade" to "Crusade", "custom" to "Custom"
    )
    ExposedDropdownMenuBox(expanded = missionPackExpanded, onExpandedChange = { missionPackExpanded = it }) {
        OutlinedTextField(
            value = missionPacks.firstOrNull { it.first == (config.missionPack ?: "") }?.second ?: "None",
            onValueChange = {},
            readOnly = true,
            label = { Text("Mission Pack") },
            trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(missionPackExpanded) },
            modifier = Modifier.fillMaxWidth().menuAnchor(MenuAnchorType.PrimaryNotEditable)
        )
        ExposedDropdownMenu(expanded = missionPackExpanded, onDismissRequest = { missionPackExpanded = false }) {
            missionPacks.forEach { (value, label) ->
                DropdownMenuItem(
                    text = { Text(label) },
                    onClick = {
                        onConfigChange(config.copy(missionPack = value))
                        missionPackExpanded = false
                    }
                )
            }
        }
    }

    AnimatedVisibility(visible = !config.missionPack.isNullOrEmpty()) {
        Column(modifier = Modifier.padding(top = 8.dp)) {
            var missionSelExpanded by remember { mutableStateOf(false) }
            val missionSelections = listOf("random" to "Random", "pre_selected" to "Pre-Selected")
            ExposedDropdownMenuBox(expanded = missionSelExpanded, onExpandedChange = { missionSelExpanded = it }) {
                OutlinedTextField(
                    value = missionSelections.firstOrNull { it.first == (config.missionSelection ?: "random") }?.second ?: "Random",
                    onValueChange = {},
                    readOnly = true,
                    label = { Text("Mission Selection") },
                    trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(missionSelExpanded) },
                    modifier = Modifier.fillMaxWidth().menuAnchor(MenuAnchorType.PrimaryNotEditable)
                )
                ExposedDropdownMenu(expanded = missionSelExpanded, onDismissRequest = { missionSelExpanded = false }) {
                    missionSelections.forEach { (value, label) ->
                        DropdownMenuItem(
                            text = { Text(label) },
                            onClick = {
                                onConfigChange(config.copy(missionSelection = value))
                                missionSelExpanded = false
                            }
                        )
                    }
                }
            }
        }
    }

    Spacer(Modifier.height(8.dp))
    OutlinedTextField(
        value = config.missionNotes ?: "",
        onValueChange = { onConfigChange(config.copy(missionNotes = it)) },
        label = { Text("Mission Notes") },
        minLines = 2,
        maxLines = 4,
        modifier = Modifier.fillMaxWidth()
    )

    // ─── Army Rules Section ───
    Spacer(Modifier.height(16.dp))
    SectionHeader("Army Rules")

    SwitchRow("Battle Ready Required", config.battleReadyRequired == true) {
        onConfigChange(config.copy(battleReadyRequired = it))
    }
    SwitchRow("WYSIWYG Required", config.wysiwygRequired == true) {
        onConfigChange(config.copy(wysiwygRequired = it))
    }
    SwitchRow("Forge World Allowed", config.forgeWorldAllowed != false) {
        onConfigChange(config.copy(forgeWorldAllowed = it))
    }
    SwitchRow("Legends Allowed", config.legendsAllowed == true) {
        onConfigChange(config.copy(legendsAllowed = it))
    }
    SwitchRow("Allow Proxies", config.allowProxies == true) {
        onConfigChange(config.copy(allowProxies = it))
    }

    AnimatedVisibility(visible = config.allowProxies == true) {
        OutlinedTextField(
            value = config.proxyNotes ?: "",
            onValueChange = { onConfigChange(config.copy(proxyNotes = it)) },
            label = { Text("Proxy Notes") },
            minLines = 2,
            maxLines = 3,
            modifier = Modifier.fillMaxWidth().padding(top = 4.dp)
        )
    }

    Spacer(Modifier.height(8.dp))
    OutlinedTextField(
        value = config.armyRulesNotes ?: "",
        onValueChange = { onConfigChange(config.copy(armyRulesNotes = it)) },
        label = { Text("Army Rules Notes") },
        minLines = 2,
        maxLines = 4,
        modifier = Modifier.fillMaxWidth()
    )

    // ─── Terrain & Table Section ───
    Spacer(Modifier.height(16.dp))
    SectionHeader("Terrain & Table")

    var terrainExpanded by remember { mutableStateOf(false) }
    val terrainTypes = listOf(
        "tournament" to "Tournament Standard", "casual" to "Casual",
        "bring_your_own" to "Bring Your Own"
    )
    ExposedDropdownMenuBox(expanded = terrainExpanded, onExpandedChange = { terrainExpanded = it }) {
        OutlinedTextField(
            value = terrainTypes.firstOrNull { it.first == (config.terrainType ?: "tournament") }?.second ?: "Tournament Standard",
            onValueChange = {},
            readOnly = true,
            label = { Text("Terrain Type") },
            trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(terrainExpanded) },
            modifier = Modifier.fillMaxWidth().menuAnchor(MenuAnchorType.PrimaryNotEditable)
        )
        ExposedDropdownMenu(expanded = terrainExpanded, onDismissRequest = { terrainExpanded = false }) {
            terrainTypes.forEach { (value, label) ->
                DropdownMenuItem(
                    text = { Text(label) },
                    onClick = {
                        onConfigChange(config.copy(terrainType = value))
                        terrainExpanded = false
                    }
                )
            }
        }
    }

    Spacer(Modifier.height(8.dp))

    var tableSizeExpanded by remember { mutableStateOf(false) }
    val tableSizes = listOf(
        "44x30" to "44\" x 30\" (Combat Patrol)",
        "44x60" to "44\" x 60\" (Standard)",
        "custom" to "Custom"
    )
    ExposedDropdownMenuBox(expanded = tableSizeExpanded, onExpandedChange = { tableSizeExpanded = it }) {
        OutlinedTextField(
            value = tableSizes.firstOrNull { it.first == (config.tableSize ?: "44x60") }?.second ?: "44\" x 60\" (Standard)",
            onValueChange = {},
            readOnly = true,
            label = { Text("Table Size") },
            trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(tableSizeExpanded) },
            modifier = Modifier.fillMaxWidth().menuAnchor(MenuAnchorType.PrimaryNotEditable)
        )
        ExposedDropdownMenu(expanded = tableSizeExpanded, onDismissRequest = { tableSizeExpanded = false }) {
            tableSizes.forEach { (value, label) ->
                DropdownMenuItem(
                    text = { Text(label) },
                    onClick = {
                        onConfigChange(config.copy(tableSize = value))
                        tableSizeExpanded = false
                    }
                )
            }
        }
    }

    // ─── Event Structure Section ───
    Spacer(Modifier.height(16.dp))
    SectionHeader("Event Structure")

    var eventTypeExpanded by remember { mutableStateOf(false) }
    val eventTypes = listOf(
        "casual" to "Casual", "tournament" to "Tournament",
        "campaign" to "Campaign", "league" to "League"
    )
    ExposedDropdownMenuBox(expanded = eventTypeExpanded, onExpandedChange = { eventTypeExpanded = it }) {
        OutlinedTextField(
            value = eventTypes.firstOrNull { it.first == (config.eventType ?: "casual") }?.second ?: "Casual",
            onValueChange = {},
            readOnly = true,
            label = { Text("Event Type") },
            trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(eventTypeExpanded) },
            modifier = Modifier.fillMaxWidth().menuAnchor(MenuAnchorType.PrimaryNotEditable)
        )
        ExposedDropdownMenu(expanded = eventTypeExpanded, onDismissRequest = { eventTypeExpanded = false }) {
            eventTypes.forEach { (value, label) ->
                DropdownMenuItem(
                    text = { Text(label) },
                    onClick = {
                        onConfigChange(config.copy(eventType = value))
                        eventTypeExpanded = false
                    }
                )
            }
        }
    }

    AnimatedVisibility(visible = config.eventType == "tournament" || config.eventType == "league") {
        Column {
            Spacer(Modifier.height(8.dp))

            // Tournament Style
            var styleExpanded by remember { mutableStateOf(false) }
            val styles = listOf(
                "swiss" to "Swiss", "single_elimination" to "Single Elimination",
                "round_robin" to "Round Robin"
            )
            ExposedDropdownMenuBox(expanded = styleExpanded, onExpandedChange = { styleExpanded = it }) {
                OutlinedTextField(
                    value = styles.firstOrNull { it.first == (config.tournamentStyle ?: "swiss") }?.second ?: "Swiss",
                    onValueChange = {},
                    readOnly = true,
                    label = { Text("Tournament Style") },
                    trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(styleExpanded) },
                    modifier = Modifier.fillMaxWidth().menuAnchor(MenuAnchorType.PrimaryNotEditable)
                )
                ExposedDropdownMenu(expanded = styleExpanded, onDismissRequest = { styleExpanded = false }) {
                    styles.forEach { (value, label) ->
                        DropdownMenuItem(
                            text = { Text(label) },
                            onClick = {
                                onConfigChange(config.copy(tournamentStyle = value))
                                styleExpanded = false
                            }
                        )
                    }
                }
            }

            Spacer(Modifier.height(8.dp))

            // Rounds
            OutlinedTextField(
                value = config.roundsCount?.toString() ?: "",
                onValueChange = { onConfigChange(config.copy(roundsCount = it.toIntOrNull())) },
                label = { Text("Rounds") },
                placeholder = { Text("Auto") },
                keyboardOptions = KeyboardOptions(keyboardType = KeyboardType.Number),
                modifier = Modifier.fillMaxWidth()
            )

            Spacer(Modifier.height(8.dp))

            // Round time
            val roundTime = (config.roundTimeMinutes ?: 150).toFloat()
            Text("Round Time: ${roundTime.toInt()} min", style = MaterialTheme.typography.bodyMedium)
            Slider(
                value = roundTime,
                onValueChange = { onConfigChange(config.copy(roundTimeMinutes = it.toInt())) },
                valueRange = 30f..300f,
                steps = 17
            )

            SwitchRow("Include Top Cut", config.includeTopCut == true) {
                onConfigChange(config.copy(includeTopCut = it))
            }

            // Scoring
            var scoringExpanded by remember { mutableStateOf(false) }
            val scoringTypes = listOf(
                "win_loss" to "Win/Loss", "win_draw_loss" to "Win/Draw/Loss",
                "battle_points" to "Battle Points"
            )
            ExposedDropdownMenuBox(expanded = scoringExpanded, onExpandedChange = { scoringExpanded = it }) {
                OutlinedTextField(
                    value = scoringTypes.firstOrNull { it.first == (config.scoringType ?: "win_loss") }?.second ?: "Win/Loss",
                    onValueChange = {},
                    readOnly = true,
                    label = { Text("Scoring") },
                    trailingIcon = { ExposedDropdownMenuDefaults.TrailingIcon(scoringExpanded) },
                    modifier = Modifier.fillMaxWidth().menuAnchor(MenuAnchorType.PrimaryNotEditable)
                )
                ExposedDropdownMenu(expanded = scoringExpanded, onDismissRequest = { scoringExpanded = false }) {
                    scoringTypes.forEach { (value, label) ->
                        DropdownMenuItem(
                            text = { Text(label) },
                            onClick = {
                                onConfigChange(config.copy(scoringType = value))
                                scoringExpanded = false
                            }
                        )
                    }
                }
            }
        }
    }

    // ─── Crusade Settings Section ───
    AnimatedVisibility(visible = config.gameType == "crusade") {
        Column {
            Spacer(Modifier.height(16.dp))
            SectionHeader("Crusade Settings")

            val supplyLimit = (config.startingSupplyLimit ?: 1000).toFloat()
            Text("Starting Supply Limit: ${supplyLimit.toInt()}", style = MaterialTheme.typography.bodyMedium)
            Slider(
                value = supplyLimit,
                onValueChange = { onConfigChange(config.copy(startingSupplyLimit = (it / 250).toInt() * 250)) },
                valueRange = 500f..3000f,
                steps = 9
            )

            val crusadePoints = (config.startingCrusadePoints ?: 0).toFloat()
            Text("Starting Crusade Points: ${crusadePoints.toInt()}", style = MaterialTheme.typography.bodyMedium)
            Slider(
                value = crusadePoints,
                onValueChange = { onConfigChange(config.copy(startingCrusadePoints = it.toInt())) },
                valueRange = 0f..20f,
                steps = 19
            )

            OutlinedTextField(
                value = config.crusadeProgressionNotes ?: "",
                onValueChange = { onConfigChange(config.copy(crusadeProgressionNotes = it)) },
                label = { Text("Progression Notes") },
                minLines = 2,
                maxLines = 4,
                modifier = Modifier.fillMaxWidth()
            )
        }
    }

    // ─── Army List Submission Section ───
    AnimatedVisibility(
        visible = config.eventType == "tournament" || config.eventType == "league"
    ) {
        Column {
            Spacer(Modifier.height(16.dp))
            SectionHeader("Army List Submission")

            SwitchRow("Require Army List", config.requireArmyList == true) {
                onConfigChange(config.copy(requireArmyList = it))
            }

            AnimatedVisibility(visible = config.requireArmyList == true) {
                Column(modifier = Modifier.padding(top = 4.dp)) {
                    OutlinedTextField(
                        value = config.armyListDeadline ?: "",
                        onValueChange = { onConfigChange(config.copy(armyListDeadline = it)) },
                        label = { Text("Submission Deadline") },
                        placeholder = { Text("YYYY-MM-DD HH:MM") },
                        modifier = Modifier.fillMaxWidth()
                    )
                    Spacer(Modifier.height(8.dp))
                    OutlinedTextField(
                        value = config.armyListNotes ?: "",
                        onValueChange = { onConfigChange(config.copy(armyListNotes = it)) },
                        label = { Text("Submission Notes") },
                        minLines = 2,
                        maxLines = 3,
                        modifier = Modifier.fillMaxWidth()
                    )
                }
            }
        }
    }

    // ─── Prizes & Entry Section ───
    Spacer(Modifier.height(16.dp))
    SectionHeader("Prizes & Entry")

    SwitchRow("Has Prizes", config.hasPrizes == true) {
        onConfigChange(config.copy(hasPrizes = it))
    }

    AnimatedVisibility(visible = config.hasPrizes == true) {
        OutlinedTextField(
            value = config.prizeStructure ?: "",
            onValueChange = { onConfigChange(config.copy(prizeStructure = it)) },
            label = { Text("Prize Structure") },
            minLines = 2,
            maxLines = 4,
            modifier = Modifier.fillMaxWidth().padding(top = 4.dp)
        )
    }

    Spacer(Modifier.height(8.dp))
    EntryFeeRow(
        entryFee = config.entryFee?.toString() ?: "",
        currency = config.entryFeeCurrency ?: "USD",
        onFeeChange = { onConfigChange(config.copy(entryFee = it.toDoubleOrNull())) },
        onCurrencyChange = { onConfigChange(config.copy(entryFeeCurrency = it)) }
    )

    Spacer(Modifier.height(8.dp))
    SwitchRow("Allow Spectators", config.allowSpectators != false) {
        onConfigChange(config.copy(allowSpectators = it))
    }
}
